doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+```
+
+As we emphasized earlier, function level binding is actually kind of inefficient, so using a bound function as a reference we can rewrite this to use component level binding instead:
+
+```python
+a_slider = pn.widgets.FloatSlider(start=0, end=10)
+b_slider = pn.widgets.FloatSlider(start=0, end=10)
+
+def add(a, b):
+    return f'{a=} + {b=} = {a+b}'
+
+pn.Column(a_slider, b_slider, pn.pane.Str(object=pn.bind(add, a_slider, b_slider))
+```
+
+## Why callbacks?
+
+So far we have expounded at length about the benefits of reactive approaches, so you might wonder: "why allow callbacks at all?" One pragmatic reason for having callbacks is that reactivity is built on top of the callback based `.param.watch` API in Param, its the lowest level API offered and lets users build higher-level APIs on top. However, from a practical perspective there's a few reasons one might use callbacks even in a scenario where reactivity drives the majority of your application.
+
+### Performance
+
+One major reason for using callbacks is again related to the fact that they are the lowest level available API. This means they have the least amount of overhead and provide the ability to perform very fine-grained updates, i.e. update the specific parameter value that should be updated.
+
+### Side Effects
+
+Another common use case for callbacks are side-effects, e.g. you may want to trigger some external action or log user interactions. In this case callbacks are the main
+
+### Transient Events
+
+Certain user interactions are, by their very nature more amenable to an event-driven approach because they are transient. One very common such example are button clicks. Since they are transient, it's not immediately obvious how to structure code to handle the event.
+
+Let us, for example, extend our simple addition app with a button that triggers the calculation:
+
+```python
+a = pn.widgets.FloatSlider()
+b = pn.widgets.FloatSlider()
+button = pn.widgets.Button(name='Calculate')
+
+str_pane = pn.pane.Str(f'{a} + {b} = {a+b}')
+
+def update(_=None):
+    str_pane.object = f'{a.value} + {b.value} = {a.value+b.value}'
+
+button.on_click(update)
+
+pn.Column(a_slider, b_slider, button, str_pane)
+```
+
+Defining a click handler using the `on_click` method to trigger the computation feels natural, but we again struggle with the fact that we have to access the state from multiple widgets. The reactive approaches therefore do provide alternatives here, e.g. when using function level binding we can determine if the button has been clicked and raise `Skip` error if it hasn't causing the update event to be skipped:
+
+```python
+a_slider = pn.widgets.FloatSlider()
+b_slider = pn.widgets.FloatSlider()
+button = pn.widgets.Button(name='Calculate')
+
+def add(a, b, compute):
+    if not compute:
+	    raise pn.param.Skip()
+    return f'{a} + {b} = {a+b}'
+
+pn.Column(a_slider, b_slider, button, pn.bind(add, a_slider, b_slider, button))
```

Comment:
You invented a new way to implement a *form*. I.e. you added `Skip` exception? I think its nice.

Is `Skip` something that should go into Param?

Reply:
I've honestly been meaning to add that for a very, very long time (several years) but always forgot. Yes it probably has to move up to param so this can be supported for bound function references.